guides/source/active_record_migrations.md

```diff
@@ -1083,32 +1352,66 @@ This will generate the following output:
+Rails keeps track of which migrations have been run through the
+`schema_migrations` table in the database. When you run a migration, Rails
+inserts a row into the `schema_migrations` table with the version number of the
+migration, stored in the `version` column. This allows Rails to determine which
+migrations have already been applied to the database.
+
+For example, if you have a migration file named 20240428000000_create_users.rb,
+Rails will extract the version number (20240428000000) from the filename and
+insert it into the schema_migrations table after the migration has been
+successfully executed.
+
+You can view the contents of the schema_migrations table directly in your
+database management tool or by using Rails console:
+
+```irb
+rails dbconsole
+```
+
+Then, within the database console, you can query the schema_migrations table:
+
+```sql
+SELECT * FROM schema_migrations;
+```
+
+This will show you a list of all migration version numbers that have been
+applied to the database. Rails uses this information to determine which
+migrations need to be run when you run rails db:migrate or rails db:migrate:up
+commands.
 
 Changing Existing Migrations
 ----------------------------
 
 Occasionally you will make a mistake when writing a migration. If you have
 already run the migration, then you cannot just edit the migration and run the
 migration again: Rails thinks it has already run the migration and so will do
-nothing when you run `bin/rails db:migrate`. You must rollback the migration (for
-example with `bin/rails db:rollback`), edit your migration, and then run
+nothing when you run `bin/rails db:migrate`. You must rollback the migration
+(for example with `bin/rails db:rollback`), edit your migration, and then run
 `bin/rails db:migrate` to run the corrected version.
 
-In general, editing existing migrations is not a good idea. You will be
-creating extra work for yourself and your co-workers and cause major headaches
-if the existing version of the migration has already been run on production
-machines.
+In general, editing existing migrations that have been already committed to
+source control is not a good idea. You will be creating extra work for yourself
+and your co-workers and cause major headaches if the existing version of the
+migration has already been run on production machines. Instead, you should write
+a new migration that performs the changes you require.
 
-Instead, you should write a new migration that performs the changes
-you require. Editing a freshly generated migration that has not yet been
-committed to source control (or, more generally, which has not been propagated
-beyond your development machine) is relatively harmless.
+However, editing a freshly generated migration that has not yet been committed
+to source control (or, more generally, which has not been propagated beyond your
+development machine) is relatively harmless.
```

Comment:
I'd even say "this is expected" instead of "relatively harmless" which could be misleading.

I personally never generate migrations that are exactly how I want. I generate them as a starting point then apply my own customizations, editing the file, then run it locally. Once it's all done I commit both files (migration file and schema dump changes) and would no longer edit from that point, instead creating a new migration as described.